### DISCUSSION TITLE
README says source not included, but it appears to be

### DISCUSSION THREAD

**Discussion by wizonesolutions** [peseoane/linux-wsl#6] [Q&A]
Is the `source` folder not the relevant kernel source? It looks like it, but maybe I'm misunderstanding something.

**Comment by peseoane** [comment]
@wizonesolutions 
> Is the `source` folder not the relevant kernel source? It looks like it, but maybe I'm misunderstanding something.

Hi!

I use the source/ folder to put the compiled kernels, in the root the readme.md and little else.

Using the source tags, all the kernels appear as it is logical, they are signed with gpg2, with string tag... 

I try to be very tidy but I accept suggestions

**Reply by wizonesolutions** [reply]
Thanks for the quick reply! Ah, so it's not even necessary to recompile it.

I have a suggestion for the installation instructions, then. WSL 2 actually allows configuring a custom kernel with the `kernel` parameter in the user's `.wslconfig`: https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configure-global-options-with-wslconfig

So, in theory, one could download your `bzImage`, save it anywhere, and then point at it in the `.wslconfig`. This worked for me, at least.

It seems like if I want to build from source, I should still download the kernel.org source, since when I tried to `make all` using the source directory of the repo, it wanted to reconfigure things even though you already include a `.config`? But since I have an Intel processor, I can most likely just use the release version.

**Reply by peseoane** [reply]
I download the versions from kernel.org, and from the M
icrosoft WLS tree I adapt the configs. This is what has worked best for me so far.

Thanks for the kernels to have other ways to test without going through PowerShell.

I need time to recover from an illness, as soon as I can I will improve the documentation. ;)

**Reply by peseoane** [reply]
Now the docs are updated, thanks! on a new system the dotfile works and is much more quick and safe.

Thank you sir.
